temporal-sdk/src/main/java/io/temporal/internal/worker/LocalActivityWorker.java

```diff
@@ -497,6 +500,10 @@ public void handle(LocalActivityAttemptTask attemptTask) throws Exception {
             processingFailed(activityTask.getActivityId(), activityTask.getAttempt(), ex));
         throw ex;
       } finally {
+        // Permit can be null in the event of a timeout while waiting on a permit
+        if (executionContext.getPermit() != null) {
```

Comment:
Really? I am suprised by this comment because how did we end up submitting the `attemptTask` without a permit?

Reply:
Yeah, I guess that's no longer true after the change to use the queue